crates/aptos-rosetta/src/common.rs

```diff
@@ -316,15 +357,27 @@ pub fn to_hex_lower<T: LowerHex>(obj: &T) -> String {
 }
 
 /// Retrieves the currency from the given parameters
-/// TODO: What do do about the type params?
-/// TODO: Handle other currencies, will need to be passed in as a config file or something on startup
-pub fn parse_currency(address: AccountAddress, module: &str, name: &str) -> ApiResult<Currency> {
-    match (address, module, name) {
-        (AccountAddress::ONE, APTOS_COIN_MODULE, APTOS_COIN_RESOURCE) => Ok(native_coin()),
-        _ => Err(ApiError::TransactionParseError(Some(format!(
-            "Invalid coin for transfer {}::{}::{}",
-            address, module, name
-        )))),
+pub fn parse_coin_currency(
```

Comment:
Any worry with there being multiple?

Reply:
This is controlled by the person running Rosetta, and the currencies are in a set.  So, if you have a duplicate, that's on the configuration.

Could check for duplicates on startup.